tests/functional/spec/list_collector.spec.js

```diff
@@ -96,6 +96,37 @@ describe('List Collector', function() {
       $(ListCollectorAddPage.submit()).click();
     });
 
+    it('The user is returned to the list collector when the cancel link is clicked.', function() {
+      $(ListCollectorPage.yes()).click();
+      $(ListCollectorPage.submit()).click();
+      $(ListCollectorAddPage.firstName()).setValue('Someone');
+      $(ListCollectorAddPage.lastName()).setValue('Else');
+      $(ListCollectorAddPage.cancelAndReturn()).click();
+      expect(browser.getUrl()).to.contain(ListCollectorPage.pageName);
+    });
+
+    it('The user is returned to the list collector when the cancel link is clicked on the edit page.', function() {
+      $(ListCollectorPage.yes()).click();
+      $(ListCollectorPage.submit()).click();
+      $(ListCollectorAddPage.firstName()).setValue('Someone');
+      $(ListCollectorAddPage.lastName()).setValue('Else');
+      $(ListCollectorAddPage.submit()).click();
+      $(ListCollectorPage.listEditLink(1)).click();
+      $(ListCollectorEditPage.cancelAndReturn()).click();
+      expect(browser.getUrl()).to.contain(ListCollectorPage.pageName);
+    });
+
+    it('The user is returned to the list collector when the cancel link is clicked on the remove page.', function() {
+      $(ListCollectorPage.yes()).click();
+      $(ListCollectorPage.submit()).click();
+      $(ListCollectorAddPage.firstName()).setValue('Someone');
+      $(ListCollectorAddPage.lastName()).setValue('Else');
+      $(ListCollectorAddPage.submit()).click();
+      $(ListCollectorPage.listRemoveLink(1)).click();
+      $(ListCollectorRemovePage.cancelAndReturn()).click();
+      expect(browser.getUrl()).to.contain(ListCollectorPage.pageName);
+    });
```

Comment:
These check the functionality of the link. We need to add the `action_text` into a schema that is not the census and make sure that is also displayed on these pages. Probably easiest to add it to the list_collector json and edit the existing integration test.

Reply:
I've now added a test for this, not sure if I need to add additional ones for the edit and remove pages too?

Reply:
I would say yes because different templates handle these.